### DISCUSSION TITLE
First click not being send via MQTT

### DISCUSSION THREAD

**Discussion by jakob42** [Koenkk/zigbee2mqtt#14691] [General]
Hi,

I switched back to z2m after a stint with a conbee II and their deconz software. Replaced the conbee with a ZBDongle-P, everything works fine and all devices are supported (they werent with deconz). But I ran into a problem that my battery powered wall switches often don't react on the first click. (I noticed it at least with the following switches: [Xiaomi WXKG02LM_rev2](https://www.zigbee2mqtt.io/devices/WXKG02LM_rev2.html#xiaomi-wxkg02lm_rev2), [TuYa TS0044](https://www.zigbee2mqtt.io/devices/TS0044.html#tuya-ts0044), [IKEA E1743](https://www.zigbee2mqtt.io/devices/E1743.html#ikea-e1743) and [Philips 324131092621](https://www.zigbee2mqtt.io/devices/324131092621.html#philips-324131092621))

```
zigbee2mqtt      | Zigbee2MQTT:debug 2022-10-28 07:58:44: Received Zigbee message from 'Schalter Wohnzimmer', type 'raw', cluster 'genOnOff', data '{"data":[1,5,253,0],"type":"Buffer"}' from endpoint 4 with groupID 0
zigbee2mqtt      | Zigbee2MQTT:info  2022-10-28 07:58:44: MQTT publish: topic 'zigbee2mqtt/Schalter Wohnzimmer', payload '{"action":null,"battery":100,"last_seen":1666936724678,"linkquality":32,"voltage":3000}'
zigbee2mqtt      | Zigbee2MQTT:debug 2022-10-28 07:58:45: Received Zigbee message from 'Schalter Wohnzimmer', type 'raw', cluster 'genOnOff', data '{"data":[1,5,253,0],"type":"Buffer"}' from endpoint 4 with groupID 0
zigbee2mqtt      | Zigbee2MQTT:info  2022-10-28 07:58:45: MQTT publish: topic 'zigbee2mqtt/Schalter Wohnzimmer', payload '{"action":null,"battery":100,"last_seen":1666936725681,"linkquality":32,"voltage":3000}'
```

`action: null` obviosly doesn't trigger anything in my HA or nodered. Half a minute later, same switch, same button:

```
zigbee2mqtt      | Zigbee2MQTT:debug 2022-10-28 07:59:25: Received Zigbee message from 'Schalter Wohnzimmer', type 'raw', cluster 'genOnOff', data '{"data":[1,6,253,0],"type":"Buffer"}' from endpoint 4 with groupID 0
zigbee2mqtt      | Zigbee2MQTT:info  2022-10-28 07:59:25: MQTT publish: topic 'zigbee2mqtt/Schalter Wohnzimmer', payload '{"action":"4_single","battery":100,"last_seen":1666936765342,"linkquality":32,"voltage":3000}'
zigbee2mqtt      | Zigbee2MQTT:info  2022-10-28 07:59:25: MQTT publish: topic 'zigbee2mqtt/Schalter Wohnzimmer', payload '{"action":"","battery":100,"last_seen":1666936765342,"linkquality":32,"voltage":3000}'
zigbee2mqtt      | Zigbee2MQTT:info  2022-10-28 07:59:25: MQTT publish: topic 'zigbee2mqtt/Schalter Wohnzimmer/action', payload '4_single'
```

The switches worked on the first click with deconz. Any idea what might be going wrong? Thanks!

**Comment by oopiicaa** [comment]
Is this first click after Home Assistant boot/reboot?

**Reply by jakob42** [reply]
No, its been up for a few days now. And IMHO this is way before HA, if I read the logs correctly.

Thanks for the reply!

**Comment by mkagawa** [comment]
I have exact same issue with TS0044 Tuya 4 button scene switch. It happens if leave button no action for 30 min or so. I guess it is related to sleep mode of the device. my version of Z2M is 1.35.1

**Comment by jakob42** [comment]
But do your get a log entry for the non working click as well? But at the moment my problem seems solved. As recommended in another thread I force removed the device, restarted z2m and added it again. My zigbee network is still sometimes very slow, but this problem seems solved to me.

**Reply by mkagawa** [reply]
hmm. I removed the tuya device completely from Z2M and re-added, but I get still same symptom.

**Comment by Breakingx** [comment]
I have the philipps hue motion sensor with the same problem. The first action is complety ignored. But waiting a minute and trying it again it works relaiable for some time until it "sleeps" again.

**Comment by jakob42** [comment]
Do you guys get anything in the log for the first skipped action like me?

**Comment by mkagawa** [comment]
I see these logs as similar as you mentioned above. With a couple of presses, it starts working again.

Debug 2024-02-26 16:54:54Received Zigbee message from 'the_Button', type 'raw', cluster 'genOnOff', data '{"data":[1,6,253,0],"type":"Buffer"}' from endpoint 4 with groupID 0
Debug 2024-02-26 16:54:55Received Zigbee message from 'the_Button', type 'raw', cluster 'genOnOff', data '{"data":[1,6,253,0],"type":"Buffer"}' from endpoint 4 with groupID 0
